org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/JavacProblemConverter.java

```diff
@@ -553,7 +553,19 @@ yield switch (rootCauseCode) {
 			case "compiler.err.not.def.public.cant.access" -> IProblem.NotVisibleType; // TODO different according to target node
 			case "compiler.err.already.defined" -> IProblem.DuplicateMethod; // TODO different according to target node
 			case "compiler.err.var.might.not.have.been.initialized" -> IProblem.UninitializedLocalVariable;
-			case "compiler.err.missing.meth.body.or.decl.abstract" -> IProblem.MethodRequiresBody;
+			case "compiler.err.missing.meth.body.or.decl.abstract" -> {
+				if (diagnostic instanceof JCDiagnostic jcDiagnostic) {
+					if (jcDiagnostic.getDiagnosticPosition() instanceof JCMethodDecl jcMethodDecl) {
+						if (jcMethodDecl.sym.enclClass() != null && jcMethodDecl.sym.enclClass().type.isInterface()) {
```

Comment:
are you sure .sym is always non-null at that point?

Reply:
Good point, better safe than sorry